### PR TITLE
Fail when trying to load frr only features in native mode

### PR DESCRIPTION
**Pull request by fedepaol**
Ideally, the checks of FRR only fields should happen in the config handler of the speaker, which is implementation aware.
This is not possible to achieve because of the defaulting mechanism in place.
Since it makes sense to have all the checks in the same place, and those checks must happen before the defaulting, we inject an implementation driven hook to check the raw config.

Current checks are meant to drop:
- ipv6 pools in case of native bgp mode
- ipv6 aggregation length
- BFD profiles

### CODE REVIEW

File: internal/config/validate_test.go [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+			config: &configFile{
+				Peers: []peer{
+					{
+						Addr: "1.2.3.4",
+					},
+				},
+			},
+			mustFail: false,
+		},
+	}
```
[the remark this answers is not in the document]
Comment (russellb) [reply internal/config/validate_test.go:93]: nevermind, you're testing "DiscordFRROnly()" explicitly ...

**Comment by russellb** [comment]
Can you take a look at the CI failures?

**Comment by fedepaol** [comment]
> Can you take a look at the CI failures?

There's definetely an issue with tests now, as we were creating always pools with v6 / v4 addresses regardless of the family we are testing, and native now will reject any bgp pool containing v6 addresses.

**Comment by fedepaol** [comment]
Should be fixed. There was another bfd flake I was not able to reproduce locally.
Need to push https://github.com/metallb/metallb/pull/1042 forward.

File: internal/config/validate.go [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+			return fmt.Errorf("peer %s has keepalive set on native bgp mode", p.Addr)
+		}
+	}
+	if len(c.BFDProfiles) > 0 {
+		return errors.New("bfd profiles section set")
+	}
+	for _, p := range c.Pools {
+		for _, a := range p.BGPAdvertisements {
+			if a.AggregationLengthV6 != nil {
+				return fmt.Errorf("pool %s has aggregation lenghtv6 was set on native bgp mode", p.Name)
```
Comment (msherif1234) [inline internal/config/validate.go:30]: typo in the error msg

**Comment by msherif1234** [comment]
very nice!! Thanks for consolidating config validation this validation function will keep growing for sure

File: internal/config/validate.go (hunk 2 above)
Comment (fedepaol) [reply internal/config/validate.go]: Good catch, I also reworked the errors so it's clear what field they refers to.

**Review by russellb — APPROVED** [review]

**Pull request merged**